core/spec/models/spree/product_spec.rb

```diff
@@ -487,4 +487,28 @@ def build_option_type_with_values(name, values)
     product = Spree::Product.new
     expect(product.master.is_master).to be true
   end
+
+  context '#discontinue!' do
+    let(:product) { create(:product, sku: 'a-sku') }
+
+    it "sets the discontinued" do
+      product.discontinue!
+      product.reload
+      expect(product.discontinued?).to be(true)
+    end
+  end
+
+  context "#discontinued?" do
+    let(:product_live) { build(:product, sku: 'a-sku') }
+    it "should be false" do
+
+      expect(product_live.discontinued?).to be(false)
+    end
+
+    let(:product_discontinued) { build(:product, sku: 'a-sku',
```

Comment:
Avoid using {...} for multi-line blocks.<br>Block body expression is on the same line as the block start.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.